### PR TITLE
Adding a local storage workflow

### PR DESCRIPTION
**Pull request by MariamFahmy98**
Adding a workflow node for the local storage engine.

### CODE REVIEW

File: doc/CONFIGURATION.md [hunk 1]
```diff
@@ -30,6 +30,7 @@ See a [sample](./etc/pgmoneta.conf) configuration for running `pgmoneta` on `loc
 | management | 0 | Int | No | The remote management port (disable = 0) |
 | compression | zstd | String | No | The compression type (none, gzip, zstd, lz4) |
 | compression_level | 3 | int | No | The compression level |
+| storage_engine | local | String | No | The storage engine type (local, ssh, s3) |
```
Comment (jesperpedersen) [inline doc/CONFIGURATION.md:33]: Remove ssh and s3 for now

File: doc/GETTING_STARTED.md [hunk 2]
```diff
@@ -35,6 +35,8 @@ base_dir = /home/pgmoneta
 
 compression = zstd
 
+storage_engine = local
```
Comment (jesperpedersen) [inline doc/GETTING_STARTED.md:38]: Add this to the tutorials as well

File: src/include/pgmoneta.h [hunk 3]
```diff
@@ -77,6 +77,10 @@ extern "C" {
 #define COMPRESSION_ZSTD 2
 #define COMPRESSION_LZ4  3
 
+#define STORAGE_ENGINE_LOCAL 0
+#define STORAGE_ENGINE_SSH   1
```
Comment (jesperpedersen) [inline src/include/pgmoneta.h:81]: Remove for now

File: src/include/pgmoneta.h [hunk 4]
```diff
@@ -77,6 +77,10 @@ extern "C" {
...
+#define STORAGE_ENGINE_S3    2
```
Comment (jesperpedersen) [inline src/include/pgmoneta.h:82]: Remove for now

File: src/libpgmoneta/configuration.c [hunk 5]
```diff
@@ -1520,6 +1534,27 @@ as_compression(char* str)
...
 
+static int
+as_storage_engine(char* str)
+{
+   if (!strcasecmp(str, "local"))
+   {
+      return STORAGE_ENGINE_LOCAL;
+   }
+
+   if (!strcasecmp(str, "ssh"))
```
Comment (jesperpedersen) [inline src/libpgmoneta/configuration.c:1545]: Remove for now

File: src/libpgmoneta/configuration.c [hunk 6]
```diff
@@ -1520,6 +1534,27 @@ as_compression(char* str)
...
+   {
+      return STORAGE_ENGINE_SSH;
+   }
+
+   if (!strcasecmp(str, "s3"))
```
Comment (jesperpedersen) [inline src/libpgmoneta/configuration.c:1550]: Remove for now

File: src/libpgmoneta/workflow.c [hunk 7]
```diff
@@ -28,6 +28,7 @@
 
 /* pgmoneta */
 #include <pgmoneta.h>
+#include <stdio.h>
```
Comment (jesperpedersen) [inline src/libpgmoneta/workflow.c:31]: Is this necessary ?

**Review by jesperpedersen — CHANGES_REQUESTED** [review]

File: src/libpgmoneta/workflow.c (hunk 7 above)
Comment (MariamFahmy98) [reply src/libpgmoneta/workflow.c]: No, I removed it.

**Comment by jesperpedersen** [comment]
Merged.

Thanks for your contribution !

**Pull request merged**